url.go

```diff
@@ -508,10 +535,12 @@ func parseHTTPURL(url *gourl.URL) (*URL, error) {
 // Examples are as follows:
 //
 //  - ch:amd64/foo-1
-//  - ch:amd64/focal/foo-1
+//  - ch:amd64/20.04/foo-1
+//  - ch:amd64/20.04:stable/foo-1
+//  - ch:amd64/20.04:stable:branch/foo-1
 //  - ch:foo-1
 //  - ch:foo
-//  - ch:amd64/focal/foo
+//  - ch:amd64/20.04/foo
```

Comment:
I don't think these examples are correct.

If you compare `ch:amd64/20.04:stable/foo-1` and `ch:amd64/windows:10/foo-1`, the middle section is _channel_ versus _name:channel_.

I think the struct should be:
```go
type URL struct {
	Schema       string // "cs", "ch" or "local".
	User         string // "joe".
	Name         string // "wordpress".
	Revision     int    // -1 if unset, N otherwise.
	Series       string // "precise" or "" if unset; "bundle" if it's a bundle.
	Base      string // "ubuntu:20.04:stable:branch" if base is present, series is empty.
	Architecture string // "amd64" or "" if unset for charmstore (v1) URLs.
}
```

Note base always has a name and track, with optional risk and branch.